tests/core/channels/test_botframework.py

```diff
@@ -0,0 +1,155 @@
+import json
+import time
+from http import HTTPStatus
+from typing import Tuple
+from unittest import mock
+from unittest.mock import Mock
+
+import freezegun
+import jwt
+import pytest
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey
+from jwt.algorithms import RSAAlgorithm
+
+from rasa.core.channels import BotFrameworkInput
+
+DAY_IN_SECONDS = 60 * 60 * 24
+
+MS_OPENID_CONFIG_RESPONSE = {
+    "issuer": "https://api.botframework.com",
+    "authorization_endpoint": "https://invalid.botframework.com",
+    "jwks_uri": "https://login.botframework.com/v1/.well-known/keys",
+    "id_token_signing_alg_values_supported": ["RS256"],
+    "token_endpoint_auth_methods_supported": ["private_key_jwt"],
+}
+
+
+@pytest.fixture(scope="function")
+def rsa_private_key() -> RSAPrivateKey:
+    return rsa.generate_private_key(
+        public_exponent=65537,
+        key_size=2048,
+    )
+
+
+@pytest.fixture(scope="function")
+def rsa_key_pair(rsa_private_key: RSAPrivateKey) -> Tuple[bytes, bytes]:
```

Comment:
Where is this fixture used? Maybe best to remove if not in use.

Reply:
Thanks, good catch, that was left over from a previous implementation of the test.